src/frontend/Shared/components/FriendList.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { connect} from 'react-redux';
+import { increment } from '../actions';
+import {GlobalState} from "../reducers";
+
+
+class FriendList extends React.Component {
+
+    collapseRight = () => {
+        let element = document.getElementById("right-panel");
+        if (element) {
+            element.style.display = "None";
+        }
+    }
+
+    openRight = () => {
+        let element = document.getElementById("right-panel");
+        if (element) {
+            element.style.display = "Block";
+        }
+    }
+
+    render() {
+        return (
+            <div>
+                <a id="right-open" className="glyphicon glyphicon-user" onClick={() => this.openRight()}/>
+          <div id="right-panel">
+            <a id="right-close" className="closebtn" onClick={() => this.collapseRight()}>&times;</a>
+            <ul id="friend-list" className="myFriend">
+              <p> Friends</p>
+                  <li>
+                  <img src={require("./dora.png")} width="30"height="30"/><span>Will</span> 
```

Comment:
You should extract this as a separate component, and take parameters like 'img-src'.